tests/subsys/usb/desc_sections/src/desc_sections.c

```diff
@@ -198,41 +198,49 @@ static void check_endpoint_allocation(struct usb_desc_header *head)
 	}
 }
 
+/* Determine the number of bytes spanned between two linker-defined
+ * symbols that are normally interpreted as pointers.
+ *
+ * Note that these symbols provide the address of packed structures,
+ * so alignment restrictions are not relevant.
+ */
+#define SYMBOL_SPAN(_ep, _sp) (ptrdiff_t)(intptr_t)((uintptr_t)(_ep) - (uintptr_t)(_sp))
+
```

Comment:
Preemptively: yes, you do need the double cast.  Both `ptrdiff_t` and `intptr_t` are signed, but may not be the same size. The inner cast is required to correctly get the signed difference of two `uintptr_t` values (which is well defined), and the outer to get a value that will produce the right answer with the `z` length modifier.

Reply:
And, as I've just realized, this shows exactly why `intptr_t` is needed.

Reply:
> And, as I've just realized, this shows exactly why intptr_t is needed.

Is it very different from what we speculated earlier?